### PR TITLE
Disable and remove `Automatic Proposition Inductives`

### PR DESCRIPTION
**Pull request by SkySkimmer**
Overlays:
- https://github.com/coq-community/coq-ext-lib/pull/150 (backwards compatible)
- https://github.com/LPCIC/coq-elpi/pull/730

### CODE REVIEW

**Comment by andres-erbsen** [comment]
I changed a `: Type` to `: Set` to silence the warning -- does this mean that doing nothing would have also worked?

File: doc/changelog/08-vernac-commands-and-options/19872-depr-auto-prop-lowering.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- **Removed:**
+  flag `Automatic Proposition Inductives` (using its effect was deprecated since 8.20)
```
Comment (jfehrle) [inline doc/changelog/08-vernac-commands-and-options/19872-depr-auto-prop-lowering.rst:2]: The option itself was not deprecated (`Set Automatic Proposition Inductives` gives no deprecation warning), nor do I see anything in the changelog or documentation saying that it was deprecated.  Shouldn't this instead be deprecated in 9.0 and then removed in 9.1 unless there's an urgent reason to remove it immediately?

Comment (SkySkimmer) [reply doc/changelog/08-vernac-commands-and-options/19872-depr-auto-prop-lowering.rst:2]: Its effect when on was deprecated. The option was there to provide the future behaviour (by unsetting the option). 
In other words relying on the option being on was deprecated. Now there is only the off behaviour.
So there is no point keeping it.

**Comment by SkySkimmer** [comment]
>I changed a : Type to : Set to silence the warning -- does this mean that doing nothing would have also worked? (I see the CI failure, but I don't understand it)
>https://github.com/mit-plv/cross-crypto/commit/433e91211a333e73566b0eaa4da1a7d12ed972aa

In this case `Type` would become really Type (ie > Set), because we rely on https://coq.inria.fr/doc/master/refman/addendum/universe-polymorphism.html#coq:flag.Universe-Minimization-ToSet to get Set and this inductive does not produce a `Set <= u` constraint to trigger the minimization.

**Comment by SkySkimmer** [comment]
This fixes a bug where
~~~coq
Polymorphic Inductive eq@{s s'|l|} {A:Type@{s|l}} (x:A) : A -> Type@{s'|_} :=
  eq_refl : eq x x.
~~~
produces `eq : ... -> Prop` (`Type@{s'|_}` is mistaken for `Type` by the lowering code)

**Comment by ppedrot** [comment]
@coqbot run full ci

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**